HIP/hip-850.md

```diff
@@ -0,0 +1,69 @@
+---
+hip: 850
+title: Enhancing Supply Key Functionality for NFT Updates in Treasury Wallet
+author: Patches (@HGraphPunks)
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Review
+created: 2024-01-02
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/660
+requested-by: HGraphPunks
+updated: 2024-01-24
+---
+
+## Abstract
+
+This HIP proposes enhancing the Supply Key functionality within the Hedera ecosystem, specifically enabling the Supply Key to update an NFT metadata field while the NFT is held in the treasury wallet. This ensures secure and controlled adaptability of NFTs, meeting dynamic needs while maintaining their integrity post-distribution.
```

Comment:
Q: why does the treasury account require this authority instead of the admin key?

Reply:
This is already how the supply key works. We're following the same flow of how users burn and re-mint tokens with the supply key, which requires treasury wallet to hold the token.

We're adding the update metadata function to the supply key, but this is not a new flow, only 1 additional function to what you can do with the supply key flow as is.

Reply:
Okay, so it's not that the treasury must be the supply key owner, it's rather the supply key owner can modify metadata when the tokens are in the treasury's balance, since it has yet to be disbursed?

Reply:
That is correct